AEFeinstein/Super-2024-Swadge-FW
# MIDI Synthesizer

### Description

Adds the MIDI player, which replaces the buzzer completely and is available system-wide. A mode can also use a standalone MIDI player instead, if it wants to do something fancier.

Still not quite complete:
* The full drumkit - most of the "normal" drums are implemented, the rest are mostly the ones that are more difficult to synthesize from scratch
* A real optimization pass over the playback code
* Anything to do with samples, which maybe we won't even need for MIDI purposes after all

I also commented out the parts of tunernome that play metronome clicks, because that used a changing `song_t`. This should be much easier to bring back in the form of a looping MIDI which can just be played back at an arbitrary tempo instead.

### Test Instructions

Play music and SFX as normal in modes. Use the USB synth mode or jukebox to play songs and SFX. Try opening and closing the quick settings menu (to test save/restore), or doing any other weird stuff. Download random MIDI files off the internet and see if they play properly.

### Ticket Links

#230 

### Readiness Checklist
